pkg/operation/seed/seed.go

```diff
@@ -53,6 +53,68 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+// NewBuilder returns a new Builder.
+func NewBuilder() *Builder {
+	return &Builder{
+		seedObjectFunc: func() (*gardencorev1beta1.Seed, error) { return nil, fmt.Errorf("seed object is required but not set") },
+	}
+}
+
+// WithSeedObject sets the seedObject attribute at the Builder.
+func (b *Builder) WithSeedObject(seedObject *gardencorev1beta1.Seed) *Builder {
+	b.seedObjectFunc = func() (*gardencorev1beta1.Seed, error) { return seedObject, nil }
+	return b
+}
+
+// WithSeedObjectFromLister sets the seedObject attribute at the Builder after fetching it from the given lister.
+func (b *Builder) WithSeedObjectFromLister(seedLister gardencorelisters.SeedLister, seedName string) *Builder {
+	b.seedObjectFunc = func() (*gardencorev1beta1.Seed, error) { return seedLister.Get(seedName) }
+	return b
+}
+
+// WithSeedSecret sets the seedSecret attribute at the Builder.
+func (b *Builder) WithSeedSecret(seedSecret *corev1.Secret) *Builder {
+	b.seedSecretFunc = func(*corev1.SecretReference) (*corev1.Secret, error) { return seedSecret, nil }
+	return b
+}
+
+// WithSeedSecretFromClient sets the seedSecret attribute at the Builder after reading it with the client.
+func (b *Builder) WithSeedSecretFromClient(ctx context.Context, c client.Client) *Builder {
```

Comment:
Can we remove this method in favor of adding a client to `Build()` and getting the secret from the client as a "default implementation", if no `seedSecretFunc` has been set, or something similar?

Reply:
The reason behind it was that we wanted to prevent that a caller has to always pass the client to `Build()` (which would be unnecessary if he already has the objects at hand and can simply pass them using `WithSeedObject`, for example.

Reply:
Sure, good point.